modules/account-lib/src/utils/helpers.ts

```diff
@@ -0,0 +1,61 @@
+import { randomBytes } from 'crypto';
+
+const _0n = BigInt(0);
+const _1n = BigInt(1);
+/**
+ * Modulo operator
+ * @description In javascript % actually gives you the reminder not modulo.
+ * @param {bigint} num
+ * @param {bigint} mod
+ * @returns {bigint}
+ */
+export function modulo(num: bigint, mod: bigint): bigint {
+  const result = num % mod;
+  // modulo operation should always return positive values for curve operations
+  return result >= _0n ? result : mod + result;
+}
+
+/**
+ * Performs modular arithmetic inverse
+ * @dev Based on https://github.com/paulmillr/noble-secp256k1/blob/97aa518b9c12563544ea87eba471b32ecf179916/index.ts#L808
+ * @param {bigint} number
+ * @param {bigint} mod
+ * @returns {bigint}
+ */
+export function modInverse(number: bigint, mod: bigint): bigint {
+  if (number === _0n || mod <= _0n) {
+    throw new Error(`invert: expected positive integers, got n=${number} mod=${mod}`);
+  }
+  // Eucledian GCD https://brilliant.org/wiki/extended-euclidean-algorithm/
+  let a = modulo(number, mod);
+  let b = mod;
+  let [x, y, u, v] = [_0n, _1n, _1n, _0n];
+  while (a !== _0n) {
+    const q = b / a;
+    const r = b % a;
+    const m = x - u * q;
+    const n = y - v * q;
+    [b, a] = [a, r];
+    [x, y] = [u, v];
+    [u, v] = [m, n];
+  }
+  const gcd = b;
+  if (gcd !== _1n) throw new Error('invert: does not exist');
+  return modulo(x, mod);
+}
+
+/**
+ * Secure random number generator
+ * generates number in the given range
+ * @param {bigint} start
+ * @param {bigint} end
+ * @param {number} size
+ * @returns {bigint}
+ */
+export function randomNumberGenerator(start: bigint, end: bigint, size = 32): bigint {
+  const value = modulo(BigInt('0x' + randomBytes(size).toString('hex')), end);
```

Comment:
Modulo generates random numbers with bias. This is not cryptographically sound.

Reply:
Updated to use the random number generator from noble which addresses this issue